### ISSUE TITLE
Disable Style/RedundantFetchBlock

### ISSUE THREAD

**Issue by searls** [testdouble/standard#527]
The rule [Style/RedundantFetchBlock](https://docs.rubocop.org/rubocop/cops_style.html#styleredundantfetchblock) is predicated on [this argument](https://github.com/fastruby/fast-ruby#hashfetch-with-argument-vs-hashfetch--block-code) that `Hash#fetch(:key) { SOME_CONSTANT }` is 3% slower than `Hash#fetch(:key, SOME_CONSTANT)`, but:

1. The rationale results in an inconsistent code style because the same analysis says that if an object is being constructed, then block is actually 50% faster because it is only constructed once
2. 3% is not a huge deal and may be even less relevant when benchmarking Ruby 3.2 or YJIT

Because the rationale for this rule is pretty weak, I propose we remove it.

**Comment by hopsoft** [comment]
This rule also causes problems for `ActiveSupport` because the method signature for [`ActiveSupport::Cache::Store#fetch`](https://github.com/rails/rails/blob/05c3f84a511663cfef265834efb7ad8d3e472682/activesupport/lib/active_support/cache.rb#L359) doesn't support the same method signature as `Hash#fetch`, so the rule actually breaks apps that use `ActiveSupport::Cache::Store` that also format with Standard. For example Standard `v1.25.1` will change this

```ruby
@cache.fetch(key) { "foo" }
```

to this

```ruby
@cache.fetch(key, "foo")
```

which breaks apps using the former pattern.

**Comment by hopsoft** [comment]
I've worked around this with the following pattern.

```
  @store.fetch(:test) do
    @logger.debug "Prevent Standard/Rubocop from jacking up the fetch block ¯\\_(ツ)_/¯"
    "value"
  end
```

**Comment by searls** [comment]
Oof. Reason enough right there.

**Comment by searls** [comment]
Released in 1.25.2 -- thanks @hopsoft!
